src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+                            DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+            }
+        }
+
+        private void UploadDiskFilesAsPageBlobs()
+        {
+            if (this.DiskFile != null)
+            {
+                var uploadedDiskBlobs = new List<Uri>();
+                var ctnName = GetPrefixBasedResourceName("cnt", false);
+                var blobNamePrefix = GetPrefixBasedResourceName("blb", false);
+                int index = 0;
+                foreach (var diskFile in this.DiskFile)
+                {
+                    PathIntrinsics currentPath = SessionState.Path;
+                    var filePath = new FileInfo(currentPath.GetUnresolvedProviderPathFromPSPath(diskFile));
+
+                    using (var vds = new VirtualDiskStream(filePath.FullName))
+                    {
+                        if (vds.DiskType == DiskType.Fixed)
+                        {
+                            long divisor = Convert.ToInt64(Math.Pow(2, 9));
+                            long rem = 0;
+                            Math.DivRem(filePath.Length, divisor, out rem);
+                            if (rem != 0)
+                            {
+                                throw new ArgumentOutOfRangeException("filePath", string.Format("Given vhd file '{0}' is a corrupted fixed vhd", filePath));
+                            }
+                        }
+                    }
+                    BlobUri destinationUri = null;
+                    BlobUri.TryParseUri(new Uri(stoAccount.PrimaryEndpoints.Blob + ctnName + "/" + blobNamePrefix + index++ + ".vhd"), out destinationUri);
+                    if (destinationUri == null)
+                    {
+                        throw new ArgumentNullException("destinationUri");
+                    }
+                    var storageCredentialsFactory = CreateStorageCredentialsFactory(destinationUri.Uri);
+                    var parameters = new UploadParameters(destinationUri, null, filePath, true, NumberOfUploaderThreads <= 0 ? DefaultNumberOfUploaderThreads : NumberOfUploaderThreads)
+                    {
+                        Cmdlet = this,
+                        BlobObjectFactory = new CloudPageBlobObjectFactory(storageCredentialsFactory, TimeSpan.FromMinutes(1))
+                    };
+                    var vhdUploadContext = VhdUploaderModel.Upload(parameters);
+                    WriteObject(vhdUploadContext);
+                    uploadedDiskBlobs.Add(vhdUploadContext.DestinationUri);
+                }
+                this.DiskLink = uploadedDiskBlobs.Select(a => a.ToString()).ToArray();
+            }
+            else
+            {
+                WriteVerbose("DiskFile input is empty or null; skip disk file uploading...");
+            }
+        }
+        
+        private StorageCredentialsFactory CreateStorageCredentialsFactory(Uri pageBlobUrl)
```

Comment:
Rather than this, why not use the existing IStorageContext facility for a generic view of storage connection properties